ds/beis_indicators/eurostat_ip/make_eurostat_ip.py

```diff
@@ -6,65 +6,128 @@
-    file = f'NUTS_{y}.xls'
-    if os.path.exists(f'{PROJECT_DIR}/data/aux/{file}')==False:
+def save_nuts_table(year, aux_path=f"{PROJECT_DIR}/data/aux/"):
+    """Saves nuts table if does not exist already
+
+    Args:
+        year (str): year for which to download nuts table for
+        aux_path (str): path to save nuts table to. Defaults to f'{PROJECT_DIR}/data/aux/'.
+    """
+    file = f"NUTS_{year}.xls"
+    if os.path.exists(f"{aux_path}{file}") == False:
         nuts = requests.get(
-            'https://ec.europa.eu/eurostat/ramon/documents/nuts/NUTS_{y}.zip')
+            f"https://ec.europa.eu/eurostat/ramon/documents/nuts/NUTS_{year}.zip"
+        )
         z = ZipFile(BytesIO(nuts.content))
-        z.extract(file,path=f'{PROJECT_DIR}/data/aux/')
-
-    nuts_table = pd.read_excel(f'{PROJECT_DIR}/data/aux/{file}')
-
-    for l in [2,3]:
-        nuts_codes[y][l]=set(
-            nuts_table.loc[(nuts_table['COUNTRY CODE']=='UK')&(
-                                nuts_table['NUTS LEVEL']==l)]['NUTS CODE'])
-
-# Collect the patent and trademark data from Eurostat
-pats = es.get_data_df('pat_ep_rtot').query("unit == 'NR'")
-trades = es.get_data_df('ipr_ta_reg')
-
-# For each NUTS codes list and name
-for nuts_level,level in zip([2,3],['nuts2','nuts3']):
-    
-    # For each table and variable name
-    for d,name in zip([pats,trades],
-                      ['epo_patent_applications','eu_trademark_applications']):
-        
-        # Extract nuts codes depending on the variable (patents are 2010)
-        if 'patent' in name:
-            nuts_list = nuts_codes['2010'][nuts_level]
-        else:
-            nuts_list = nuts_codes['2013'][nuts_level]
-            
-        # Select the data. We will focus on activity after 2005
-        sel = d.loc[
-            [x in nuts_list for x in d['geo\\time']]].reset_index(
-            drop=True).drop('unit',1).melt(id_vars='geo\\time').query(
-            "variable > 2005")
-        
-        # Make indicator
-        if 'patent' in name:
-            nuts_spec =2010
-        else:
-            nuts_spec =2013
-
-        ind = make_indicator(sel,{'value':name},year_var='variable',nuts_var='geo\\time',
-                    nuts_spec=nuts_spec)
-    
-        logging.info(str(min(ind['year'])))
-        logging.info(str(max(ind['year'])))
-        
-        # Save indicator
-        save_indicator(ind,TARGET_PATH,f"{name}.{level}")
-    
\ No newline at end of file
+        z.extract(file, path=aux_path)
+
+
+def set_nuts_codes(year):
+    """Set nuts2 and nuts3 country codes values in dictionary
+
+    Args:
+        year (str): year to set values for
+    """
+    nuts_table = pd.read_excel(f"{PROJECT_DIR}/data/aux/NUTS_{year}.xls")
+    for l in [2, 3]:
+        nuts_codes[year][l] = set(
+            nuts_table.loc[
+                (nuts_table["COUNTRY CODE"] == "UK") & (nuts_table["NUTS LEVEL"] == l)
+            ]["NUTS CODE"]
+        )
+
+
+def create_nuts_list_spec(name):
```

Comment:
I think a better way to set the spec year is using [this function](https://github.com/nestauk/beis-indicators/blob/dev/ds/beis_indicators/geo/nuts.py#L117) . Or was the NUTS spec years specified in the dataset?

Reply:
I got the NUTS spec from make_eurostat_ip.py where it said patents should use 2010 and trademarks should use 2013.
Does that function also work for nuts3? It looks like it is for nuts2?

Reply:
@Jack-0-0 Okay, if that's the case, I'll approve the PR! 

You're right, it's only for nuts2.